src/OrchardCore.Modules/OrchardCore.Markdown/Drivers/MarkdownBodyPartDisplay.cs

```diff
@@ -63,16 +74,21 @@ public override async Task<IDisplayResult> UpdateAsync(MarkdownBodyPart model, I
             return Edit(model, context);
         }
 
-        private async ValueTask BuildViewModel(MarkdownBodyPartViewModel model, MarkdownBodyPart MarkdownBodyPart)
+        private async ValueTask BuildViewModel(MarkdownBodyPartViewModel model, MarkdownBodyPart markdownBodyPart, MarkdownBodyPartSettings settings)
         {
-            model.Markdown = MarkdownBodyPart.Markdown;
-            model.MarkdownBodyPart = MarkdownBodyPart;
-            model.ContentItem = MarkdownBodyPart.ContentItem;
+            model.Markdown = markdownBodyPart.Markdown;
+            model.MarkdownBodyPart = markdownBodyPart;
+            model.ContentItem = markdownBodyPart.ContentItem;
 
-            model.Markdown = await _liquidTemplateManager.RenderAsync(MarkdownBodyPart.Markdown, _htmlEncoder, model,
-                scope => scope.SetValue("ContentItem", model.ContentItem));
+            if (settings.AllowCustomScripts)
+            {
+                model.Markdown = await _liquidTemplateManager.RenderAsync(markdownBodyPart.Markdown, _htmlEncoder, model,
+                    scope => scope.SetValue("ContentItem", model.ContentItem));
+            }
 
+            model.Markdown = await _safeCodeFilterManager.ProcessAsync(model.Markdown ?? "");
             model.Html = Markdig.Markdown.ToHtml(model.Markdown ?? "");
+            model.Html = _htmlScriptSanitizer.Sanitize(model.Html ?? "");
         }
```

Comment:
for now I have done sanitization on Display, just to get a start on this. If we can find a good way to move it to Edit time, that would be great